packages/contracts/test/upgrade/dao.ts

```diff
@@ -0,0 +1,400 @@
+        daoCurrentImplementaion.address,
+        DAO_Current.interface.encodeFunctionData('initializeFrom', [
+          [1, 0, 0],
+          EMPTY_DATA,
+        ])
+      );
+
+      // Check that the stored implementatio has changed.
+      const implementationAfterUpgrade = await readImplementationValueFromSlot(
+        daoV100Proxy.address
+      );
+      expect(implementationAfterUpgrade).to.equal(
+        daoCurrentImplementaion.address
+      );
+      expect(implementationAfterUpgrade).to.not.equal(daoV100Implementation);
+
+      // Check that the old forwarder is still unchanged.
+      expect(await daoV100Proxy.getTrustedForwarder()).to.equal(FORWARDER_1);
+
+      // Create an action to change the forwarder to a new address.
+      const testAction = {
+        to: daoV100Proxy.address,
+        data: daoV100Proxy.interface.encodeFunctionData('setTrustedForwarder', [
+          FORWARDER_2,
+        ]),
+        value: 0,
+      };
+
+      // Execute and check in the event that the forwarder1 has been set.
+      await expect(daoV100Proxy.execute(ZERO_BYTES32, [testAction], 0))
+        .to.emit(daoV100Proxy, 'TrustedForwarderSet')
+        .withArgs(FORWARDER_2);
+
+      // Check that the storage variable is now forwarder 2.
+      expect(await daoV100Proxy.getTrustedForwarder()).to.equal(FORWARDER_2);
+    });
+  });
+
+  context(`v1.2.0 to v1.3.0`, function () {
+    beforeEach(async function () {
+      daoV120Proxy = await deployWithProxy(DAO_V1_2_0);
+      await daoV120Proxy.initialize(
+        DUMMY_METADATA,
+        signers[0].address,
+        ethers.constants.AddressZero,
+        daoExampleURI
+      );
+
+      // Store the v1.2.0 implementation
+      daoV120Implementation = await readImplementationValueFromSlot(
+        daoV120Proxy.address
+      );
+
+      // Grant the upgrade permission
+      await daoV120Proxy.grant(
+        daoV120Proxy.address,
+        signers[0].address,
+        UPGRADE_PERMISSIONS.UPGRADE_DAO_PERMISSION_ID
+      );
+    });
+
+    it('does not corrupt the DAO storage', async () => {
+      // Upgrade and call `initializeFrom`.
+      const upgradeTx = await daoV120Proxy.upgradeToAndCall(
+        daoCurrentImplementaion.address,
+        DAO_Current.interface.encodeFunctionData('initializeFrom', [
+          [1, 2, 0],
+          EMPTY_DATA,
+        ])
+      );
+
+      // Check the stored implementation.
+      const implementationAfterUpgrade = await readImplementationValueFromSlot(
+        daoV120Proxy.address
+      );
+      expect(implementationAfterUpgrade).to.equal(
+        daoCurrentImplementaion.address
+      );
+      expect(implementationAfterUpgrade).to.not.equal(daoV120Implementation);
+
+      // Check the emitted implementation.
+      const emittedImplementation = (
+        await findEventTopicLog(upgradeTx, DAO_V1_2_0.interface, 'Upgraded')
+      ).args.implementation;
+      expect(emittedImplementation).to.equal(daoCurrentImplementaion.address);
```

Comment:
Is the DAO URI at the end of the storage slots? Is there any other storage variable even further?